### ISSUE TITLE
IncompatibleClassChangeError when using "boot repl"

### ISSUE THREAD

**Issue by marcpaterno** [boot-clj/boot#482]
Using `boot repl` results in a failure:
<pre>
$ boot -vvv repl
Acquired java.util.concurrent.Semaphore@7cc63800[Permits = 0]...
Committing fileset...
Commit complete.
Released java.util.concurrent.Semaphore@7cc63800[Permits = 1]...
nREPL server started on port 49421 on host 127.0.0.1 - nrepl://127.0.0.1:49421
Acquired java.util.concurrent.Semaphore@555f202b[Permits = 0]...
Syncing project dirs to temp dirs...
Sync complete.
Released java.util.concurrent.Semaphore@555f202b[Permits = 1]...
REPL-y 0.3.7, nREPL 0.2.12
Clojure 1.8.0
Java HotSpot(TM) 64-Bit Server VM 1.8.0_92-b14
        Exit: Control+D or (exit) or (quit)
    Commands: (user/help)
        Docs: (doc function-name-here)
              (find-doc "part-of-name-here")
Find by Name: (find-name "part-of-name-here")
      Source: (source function-name-here)
     Javadoc: (javadoc java-object-or-class-here)
    Examples from clojuredocs.org: [clojuredocs or cdoc]
              (user/clojuredocs name-here)
              (user/clojuredocs "ns-here" "name-here")
IncompatibleClassChangeError Found class jline.Terminal, but interface was expected
	jline.console.ConsoleReader.<init> (ConsoleReader.java:230)
	jline.console.ConsoleReader.<init> (ConsoleReader.java:221)
	jline.console.ConsoleReader.<init> (ConsoleReader.java:213)
	reply.reader.simple-jline/setup-console-reader (simple_jline.clj:71)
	reply.reader.simple-jline/get-input-line (simple_jline.clj:109)
	clojure.lang.Atom.swap (Atom.java:37)
	clojure.core/swap! (core.clj:2260)
	clojure.core/swap! (core.clj:2253)
	reply.reader.simple-jline/safe-read-line (simple_jline.clj:148)
	reply.reader.simple-jline/safe-read-line (simple_jline.clj:155)
	clojure.core/partial/fn--4759 (core.clj:2515)
	reply.parsing/parsed-forms (parsing.clj:60)
Bye for now!
</pre>

I am running on OSX Yosemite, with what I believe to be the most recent release of boot:
<pre>
$ boot --version
#http://boot-clj.com
#Thu Jul 21 17:52:30 EDT 2016
BOOT_CLOJURE_NAME=org.clojure/clojure
BOOT_CLOJURE_VERSION=1.8.0
BOOT_VERSION=2.6.0
</pre>

Any suggestions for how to debug this would be welcome. In case it is relevant, running `lein repl` works OK.

**Comment by martinklepsch** [comment]
@marcpaterno do you have a local `build.boot`, or a `~/.boot/profile.boot` that might be relevant? Also if you installed boot for the first time a long time ago `rm -rf ~/.boot/cache` has solved many issues 😛

**Comment by marcpaterno** [comment]
I have no local `build.boot`; I am trying to run the REPL in an empty directory. I have removed my `~/.boot/cache` directory. I have no `~/.boot/profile` file.

However, I found through using `boot show --env` that there were two `jline` jar files installed in ` /Library/Java/Extensions/`. I have deleted both of them, and now the `boot repl` is working. I think this issue can be closed. I hope this information is of use to others!
